### DISCUSSION TITLE
关于使用vemss 中转住宅ip

### DISCUSSION THREAD

**Discussion by mm200990** [v2fly/v2ray-core#2504] [Q&A]
我有个vps:A  设置了vmess,  
买了个落地机（住宅ip）,我不能对该服务器进行设置, 我只有ip 地址，用户名 密码，端口  信息
目标
我想 使用v2ray客户端 连接vpsA 的vmess，再跳转落地机连外网
客户端-----vpsA (vmess)----住宅ip----外网

在单用vmess的时候，我是可以正常上外网的，

一下为我的客户端客户信息，客户端没有任何报错信息，跟平常一样，我不知道是不是哪里设置不对了，
{
  "log": {
    "access": "",
    "error": "",
    "loglevel": "warning"
  },
  "inbounds": [
    {
      "tag": "socks",
      "port": 10808,
      "listen": "127.0.0.1",
      "protocol": "socks",
      "sniffing": {
        "enabled": true,
        "destOverride": [
          "http",
          "tls"
        ],
        "routeOnly": false
      },
      "settings": {
        "auth": "noauth",
        "udp": true,
        "allowTransparent": false
      }
    }
    
  ],
  "outbounds": [
    {
    "tag": "proxy",
      "protocol": "vmess",
      "settings": {
        "vnext": [
          {
            "address": "xxx.xl",
            "port": 123,
            "users": [
              {
                "id": "xxxxxxxxxxxxxx",
                "alterId": 0,
                "email": "t@t.tt",
                "security": "auto"
              }
            ]
          }
        ]
      },
      "streamSettings": {
        "network": "ws",
        "security": "tls",
        "tlsSettings": {
          "allowInsecure": false,
          "show": false
        },
        "wsSettings": {
          "path": "/xxx",
          "headers": {}
        }
      },
      "mux": {
        "enabled": false,
        "concurrency": -1
      },
    
      "proxySettings": {
          "tag": "transit"  
        }   
        
        
    },
    
       {
      
      "tag": "transit",
      "protocol": "http",
      "settings": {
        "servers": [
            {
              "address": "xxx.xxx.xx.xxx",
              "port": 1111,
              "users": [
                {
                  "username": "xxxx",
                  "password": "xxxx" 
                }
              ] 
            }
          ]
        
        
      }
     
    },
    
    
    
    
    
    
    
    {
      "tag": "direct",
      "protocol": "freedom",
      "settings": {}
    },
    {
      "tag": "block",
      "protocol": "blackhole",
      "settings": {
        "response": {
          "type": "http"
        }
      }
    }
  ],
    "routing": {
    "domainStrategy": "AsIs",
    "rules": [
      {
        "type": "field",
        "inboundTag": [
          "api"
        ],
        "outboundTag": "api",
        "enabled": true
      },
      {
        "id": "5185765808990946219",
        "type": "field",
        "outboundTag": "direct",
        "domain": [
          "domain:example-example.com",
          "domain:example-example2.com"
        ],
        "enabled": true
      },
      {
        "id": "5515817881725590898",
        "type": "field",
        "outboundTag": "block",
        "domain": [
          "geosite:category-ads-all"
        ],
        "enabled": true
      },
      {
        "id": "5349453818422983942",
        "type": "field",
        "outboundTag": "direct",
        "domain": [
          "geosite:cn"
        ],
        "enabled": true
      },
      {
        "id": "4799385740686860411",
        "type": "field",
        "outboundTag": "direct",
        "ip": [
          "geoip:private",
          "geoip:cn"
        ],
        "enabled": true
      },
      {
        "id": "5018108784003535192",
        "type": "field",
        "port": "0-65535",
        "outboundTag": "proxy",
        "enabled": true
      }
    ]
  }
}

一下为客户端输出信息，没有报错，不知道哪有问题，
2023/05/10 20:45:06 tcp:127.0.0.1:54063 accepted tcp:alive.github.com:443 [proxy]
2023/05/10 20:45:07 tcp:127.0.0.1:54066 accepted tcp:beacons.gvt2.com:443 [proxy]
2023/05/10 20:45:12 tcp:127.0.0.1:54069 accepted tcp:assets.msn.com:443 [proxy]
2023/05/10 20:45:23 tcp:127.0.0.1:54074 accepted tcp:ds.kaspersky.com:443 [proxy]
2023/05/10 20:45:36 tcp:127.0.0.1:54079 accepted tcp:www.google.com:443 [proxy]

请大师帮忙下把，谢谢

**Comment by mydogshitgold** [comment]
The configuration part is not very readable, can you reformat it with markdown code block?

**Reply by mm200990** [reply]
sorry about that! now , i have done with the formating!  could you please have a look on it！ Without the other server, the VMESS is working fine.  i got no error message while using the following code to connect the VMESS and redirect the traffic to that "STATIC IP" server .
client---- VMESS server----Static IP server----netflix
Actually im using the method to unblock netflix, Vmess cant access Netflix, however , Static IP server is able to access the Netflix


  {
          "log": {
          "access": "",
          "error": "",
          "loglevel": "warning"
        },
     
        "inbounds": [
           {
              "tag": "socks",
               "port": 10808,
               "listen": "127.0.0.1",
                "protocol": "socks",
                "sniffing": {
                  "enabled": true,
                   "destOverride": [
                     "http",
                     "tls"
                    ],
                   "routeOnly": false
                },
                "settings": {
                 "auth": "noauth",
                 "udp": true,
                 "allowTransparent": false
                }
              
            }
            
        ],
        
        
        "outbounds": [
            {
             "tag": "proxy",
             "protocol": "vmess",
                "settings": { 
                    "vnext": [
                        {
                            "address": "xxxxxx.xx",
                             "port": 501,
                               "users": [
                                    {
                                     "id": "xx-xxxx-xxxx-xxxx-xxxx",
                                     "alterId": 0,
                                     "email": "t@t.tt",
                                     "security": "auto"
                                    }
                                ]
                        }
                    ]
                },
             
               "streamSettings": {
                     "network": "ws",
                     "security": "tls",
                        "tlsSettings": {
                         "allowInsecure": false,
                         "show": false
                        },
                     
                     "wsSettings": {
                     "path": "/ray",
                     "headers": {}
                     }
                },
             
               "mux": {
                 "enabled": false,
                 "concurrency": -1
                },
                 
                "proxySettings": {
                 "tag": "transit"  
                }   
            },
            
            
           {
             "tag": "transit",
             "protocol": "http",
               "settings": {
                   "servers": [
                        {
                         "address": "xxx.xxx.xx.xx",
                         "port": "xxxx",
                            "users": [
                                {
                                 "username": "xxxx",
                                 "password": "xxxxxxx" 
                                }
                            ] 
                        }
                    ]
                },
              
             "streamSettings": {
                 "security": "tls",
                     "tlsSettings": {
                     "allowInsecure": false
                    }
                }
            },
         
         
            {
             "tag": "direct",
             "protocol": "freedom",
             "settings": {}
            },
          
          
            {
             "tag": "block",
             "protocol": "blackhole",
                "settings": {
                    "response": {
                     "type": "http"
                    }
               }
            }
         
        ],
        
        
      
       "routing": {
         "domainStrategy": "AsIs",
           "rules": [
                {
                 "type": "field",
                    "inboundTag": [
                      "api"
                    ],
                 "outboundTag": "api",
                 "enabled": true
                },
                
                
                {
                 "id": "5185765808990946219",
                 "type": "field",
                 "outboundTag": "direct",
                    "domain": [
                      "domain:example-example.com",
                      "domain:example-example2.com"
                    ],
                 "enabled": true
                },
                
                {
                 "id": "5515817881725590898",
                 "type": "field",
                 "outboundTag": "block",
                 "domain": [
                    "geosite:category-ads-all"
                     ],
                 "enabled": true
                },
                
                
                {
                 "id": "5349453818422983942",
                 "type": "field",
                 "outboundTag": "direct",
                 "domain": [
                    "geosite:cn"
                     ],
                 "enabled": true
                },
                
                
                {
                 "id": "4799385740686860411",
                 "type": "field",
                 "outboundTag": "direct",
                 "ip": [
                    "geoip:private",
                     "geoip:cn"
                    ],
                 "enabled": true
                },
                
                
                {
                 "id": "5018108784003535192",
                 "type": "field",
                 "port": "0-65535",
                 "outboundTag": "proxy",
                 "enabled": true
                }
            ]
        }
    }

**Reply by mm200990** [reply]
even though， with the above code, no error message , but i fail to access google, youtube, etc!

**Comment by nursery01** [comment]


**Reply by mm200990** [reply]
不是的，是香港的，其实我就是想解锁奈菲，折腾了很久，没搞到，client 是国内，住宅是香港，vemes是台湾， 我上面的配置文件，放在v2ray客户端跑，没有出错，但死活连不上，不知道为什么，
连接客户端显示这个
2023/05/10 20:45:06 tcp:127.0.0.1:54063 accepted tcp:alive.github.com:443 [proxy]
2023/05/10 20:45:07 tcp:127.0.0.1:54066 accepted tcp:beacons.gvt2.com:443 [proxy]
2023/05/10 20:45:12 tcp:127.0.0.1:54069 accepted tcp:assets.msn.com:443 [proxy]
2023/05/10 20:45:23 tcp:127.0.0.1:54074 accepted tcp:ds.kaspersky.com:443 [proxy]
2023/05/10 20:45:36 tcp:127.0.0.1:54079 accepted tcp:www.google.com:443 [proxy]

不知道到底出现什么问题， 试过找文档，都试了一遍，连上了不报错，就是上不了外网

**Reply by MGX-LJY** [reply]
根据您的需求和提供的配置文件，您希望使用以下路径访问外部网络：

Copy
客户端 ----> VPS（vmess）----> 住宅IP（HTTP代理）----> 外网
在您的配置文件中，您已经将proxySettings设置为transit，这是正确的。但是，您需要在routing规则中添加一个规则，以便将出站流量从proxy重定向到transit。请将以下规则添加到routing部分的rules列表中：

json
Copy
{
  "type": "field",
  "inboundTag": ["socks"],
  "outboundTag": "transit"
}
然后，将原有的以下规则从rules列表中删除：

json
Copy
{
  "id": "5018108784003535192",
  "type": "field",
  "port": "0-65535",
  "outboundTag": "proxy",
  "enabled": true
}
在进行了上述更改后，V2Ray客户端应该能够将流量从vmess代理（VPS A）重定向到HTTP代理（住宅IP）。如果仍然无法正常工作，请确保您的住宅IP代理服务器可以正常工作，并检查VPS A上的vmess服务器配置是否允许进行代理和中转操作。

**Reply by MGX-LJY** [reply]
抱歉刚才的回答没有解决您的问题。 经过仔细查看您的配置文件，我发现了一个问题：在代理链的设置中，您的 proxy 出站设置使用了 proxySettings，而在 transit 出站设置中使用了 HTTP 代理。根据您的需求，您应该在 proxy 出站设置中直接使用 HTTP 代理。请按照以下步骤修改您的配置文件：

删除 proxy 出站设置中的 proxySettings 部分：
json
Copy
"proxySettings": {
  "tag": "transit"
}
在 proxy 出站设置中添加 HTTP 代理设置：
json
Copy
"httpSettings": {
  "servers": [
    {
      "address": "xxx.xxx.xx.xxx",
      "port": 1111,
      "users": [
        {
          "username": "xxxx",
          "password": "xxxx"
        }
      ]
    }
  ]
}
删除完整的 transit 出站设置，因为它不再需要。

更新路由规则，将流量从 socks 入站重定向到 proxy 出站：

json
Copy
{
  "type": "field",
  "inboundTag": ["socks"],
  "outboundTag": "proxy"
}
删除原有的以下规则：

json
Copy
{
  "id": "5018108784003535192",
  "type": "field",
  "port": "0-65535",
  "outboundTag": "proxy",
  "enabled": true
}
现在，配置文件应该将您的流量从客户端通过 VPS（vmess）发送到 HTTP 代理（住宅IP），然后通过 HTTP 代理访问外部网络。如果问题仍然存在，请确保您的 HTTP 代理服务器（住宅IP）正常工作，并检查 vmess 服务器配置是否正确。

**Comment by mm200990** [comment]
我有个vps:A 设置了vmess,
买了个落地机（住宅ip）,我不能对该服务器进行设置, 我只有ip 地址，用户名 密码，端口 信息
目标
我想 使用v2ray客户端 连接vpsA 的vmess，再跳转落地机连外网
客户端-----vpsA (vmess)----住宅ip----外网

在单用vmess的时候，我是可以正常上外网的，

一下为我的客户端客户信息，客户端没有任何报错信息，跟平常一样，我不知道是不是哪里设置不对了，



 {
          "log": {
          "access": "",
          "error": "",
          "loglevel": "warning"
        },
     
        "inbounds": [
           {
              "tag": "socks",
               "port": 10808,
               "listen": "127.0.0.1",
                "protocol": "socks",
                "sniffing": {
                  "enabled": true,
                   "destOverride": [
                     "http",
                     "tls"
                    ],
                   "routeOnly": false
                },
                "settings": {
                 "auth": "noauth",
                 "udp": true,
                 "allowTransparent": false
                }
              
            }
            
        ],
        
        
        "outbounds": [
            {
             "tag": "proxy",
             "protocol": "vmess",
                "settings": { 
                    "vnext": [
                        {
                            "address": "xxxxxx.xx",
                             "port": 501,
                               "users": [
                                    {
                                     "id": "xx-xxxx-xxxx-xxxx-xxxx",
                                     "alterId": 0,
                                     "email": "t@t.tt",
                                     "security": "auto"
                                    }
                                ]
                        }
                    ]
                },
             
               "streamSettings": {
                     "network": "ws",
                     "security": "tls",
                        "tlsSettings": {
                         "allowInsecure": false,
                         "show": false
                        },
                     
                     "wsSettings": {
                     "path": "/ray",
                     "headers": {}
                     }
                },
             
               "mux": {
                 "enabled": false,
                 "concurrency": -1
                },
                 
                "proxySettings": {
                 "tag": "transit"  
                }   
            },
            
            
           {
             "tag": "transit",
             "protocol": "http",
               "settings": {
                   "servers": [
                        {
                         "address": "xxx.xxx.xx.xx",
                         "port": "xxxx",
                            "users": [
                                {
                                 "username": "xxxx",
                                 "password": "xxxxxxx" 
                                }
                            ] 
                        }
                    ]
                },
              
             "streamSettings": {
                 "security": "tls",
                     "tlsSettings": {
                     "allowInsecure": false
                    }
                }
            },
         
         
            {
             "tag": "direct",
             "protocol": "freedom",
             "settings": {}
            },
          
          
            {
             "tag": "block",
             "protocol": "blackhole",
                "settings": {
                    "response": {
                     "type": "http"
                    }
               }
            }
         
        ],
        
        
      
       "routing": {
         "domainStrategy": "AsIs",
           "rules": [
                {
                 "type": "field",
                    "inboundTag": [
                      "api"
                    ],
                 "outboundTag": "api",
                 "enabled": true
                },
                
                
                {
                 "id": "5185765808990946219",
                 "type": "field",
                 "outboundTag": "direct",
                    "domain": [
                      "domain:example-example.com",
                      "domain:example-example2.com"
                    ],
                 "enabled": true
                },
                
                {
                 "id": "5515817881725590898",
                 "type": "field",
                 "outboundTag": "block",
                 "domain": [
                    "geosite:category-ads-all"
                     ],
                 "enabled": true
                },
                
                
                {
                 "id": "5349453818422983942",
                 "type": "field",
                 "outboundTag": "direct",
                 "domain": [
                    "geosite:cn"
                     ],
                 "enabled": true
                },
                
                
                {
                 "id": "4799385740686860411",
                 "type": "field",
                 "outboundTag": "direct",
                 "ip": [
                    "geoip:private",
                     "geoip:cn"
                    ],
                 "enabled": true
                },
                
                
                {
                 "id": "5018108784003535192",
                 "type": "field",
                 "port": "0-65535",
                 "outboundTag": "proxy",
                 "enabled": true
                }
            ]
        }
    }

**Reply by MGX-LJY** [reply]
根据您的需求，我提供了一个修改后的配置文件。这个配置文件将 V2Ray 客户端从 vmess 代理（VPS A）重定向到 HTTP 代理（住宅IP），然后访问外部网络。请注意，我假设您的 HTTP 代理支持连接到 VPS A 的 TLS。

json
Copy
{
  "log": {
    "access": "",
    "error": "",
    "loglevel": "warning"
  },
  "inbounds": [
    {
      "tag": "socks",
      "port": 10808,
      "listen": "127.0.0.1",
      "protocol": "socks",
      "sniffing": {
        "enabled": true,
        "destOverride": [
          "http",
          "tls"
        ],
        "routeOnly": false
      },
      "settings": {
        "auth": "noauth",
        "udp": true,
        "allowTransparent": false
      }
    }
  ],
  "outbounds": [
    {
      "tag": "proxy",
      "protocol": "vmess",
      "settings": {
        "vnext": [
          {
            "address": "xxxxxx.xx",
            "port": 501,
            "users": [
              {
                "id": "xx-xxxx-xxxx-xxxx-xxxx",
                "alterId": 0,
                "email": "t@t.tt",
                "security": "auto"
              }
            ]
          }
        ]
      },
      "streamSettings": {
        "network": "ws",
        "security": "tls",
        "tlsSettings": {
          "allowInsecure": false,
          "show": false
        },
        "wsSettings": {
          "path": "/ray",
          "headers": {}
        }
      },
      "mux": {
        "enabled": false,
        "concurrency": -1
      },
      "proxySettings": {
        "tag": "transit"
      }
    },
    {
      "tag": "transit",
      "protocol": "http",
      "settings": {
        "servers": [
          {
            "address": "xxx.xxx.xx.xx",
            "port": "xxxx",
            "users": [
              {
                "username": "xxxx",
                "password": "xxxxxxx"
              }
            ]
          }
        ]
      },
      "streamSettings": {
        "security": "tls",
        "tlsSettings": {
          "allowInsecure": false
        }
      }
    },
    {
      "tag": "direct",
      "protocol": "freedom",
      "settings": {}
    },
    {
      "tag": "block",
      "protocol": "blackhole",
      "settings": {
        "response": {
          "type": "http"
        }
      }
    }
  ],
  "routing": {
    "domainStrategy": "AsIs",
    "rules": [
      {
        "type": "field",
        "inboundTag": [
          "api"
        ],
        "outboundTag": "api",
        "enabled": true
      },
      {
        "id": "5185765808990946219",
        "type": "field",
        "outboundTag": "direct",
        "domain": [
          "domain:example-example.com",
          "domain:example-example2.com"
        ],
        "enabled": true
      },
      {
        "id": "5515817881725590898",
        "type": "field",
        "outboundTag": "block",
        "domain": [
          "geosite:category-ads-all"
        ],
        "enabled": true
      },
      {
        "id": "5349453818422983942",
        "type": "field",
        "outboundTag": "direct",
        "domain": [
          "geosite:cn"
        ],
        "enabled": true
      },
      {
        "id": "4799385740686860411",
        "type": "field",
        "outboundTag": "direct",
        "ip": [
          "geoip:private",
          "geoip:cn"
        ],
        "enabled": true
      },
      {
        "type": "field",
        "inboundTag": [
          "socks"
        ],
        "outboundTag": "proxy",
        "enabled": true
      }
    ]
  }
}
在这个配置中，我移除了之前的 "port" 规则，这样所有从 "socks" 入站的流量将直接传递给 "proxy" 出站。此外，我保留了您的 "proxySettings"，它将 "proxy" 出站的流量重定向到 "transit" 出站（即 HTTP 代理，即住宅 IP）。这样，您的流量将通过 VPS A 的 vmess 代理，然后通过住宅 IP 的 HTTP 代理，最后到达外部网络。

请尝试使用这个修改后的配置文件，看看是否能满足您的需求。如果仍然无法正常工作，请查看 V2Ray 的日志以获取可能的错误信息。您可能需要调整日志级别以获取更详细的信息。

另外，根据您的需求和网络环境，您可能需要进一步调整路由规则以满足特定的访问策略。例如，您可能希望特定的网站或服务通过 "proxy" 出站，而其他流量直接通过 "direct" 出站访问。这可以通过在 "routing" 部分添加更多的 "field" 规则来实现。

**Comment by qianbinbin** [comment]
搜了一下，所谓的住宅 IP 原来就是个 socks 代理（好像你的是 http 代理？）
我自己没用过，理论上在 VPS 添加一条 outbound 到你的住宅 IP，路由规则设置一下哪些域名走住宅 IP，就可以了
另外，如果你本地是透明代理，直接设置住宅 IP 代理应该也可以

**Reply by Mr250luch** [reply]
礼问下这边不设规则，默认出站全部走走住宅ip的sokcs代理，能连通但没法上网，设置分流规则指定域名走住宅ip的socks代理，然后追加一条freedom出站后，能正常上网了并且指定的域名确实是走了住宅ip。。但是我想所有的流量都走住宅ip的sokcs代理不知道是不是出站规则有问题？

**Reply by qianbinbin** [reply]
不知道你那边是什么情况，注意一下tag有没有设置正确，把log级别开到最大看看

**Reply by Mr250luch** [reply]
看了下。。debug里一直重复这几段错误。。(xxxx是我的sokcs5 住宅IP）

2023/07/12 14:37:12 [Debug] transport/internet: dialing to tcp:xxx.xxx.xxx.xxx:xxxx
2023/07/12 14:37:14 [Info] [1321903743] app/proxyman/inbound: connection ends > proxy/vless/inbound: connection ends > proxy/vless/inbound: failed to transfer request payload > websocket: close 1000 (normal)
2023/07/12 14:37:41 [Info] [3639962393] app/proxyman/inbound: connection ends > proxy/vless/inbound: connection ends > EOF
2023/07/12 14:37:43 [Info] [4145505862] proxy/vless/inbound: firstLen = 271
2023/07/12 14:37:43 [Info] [4145505862] proxy/vless/inbound: received request for udp:8.8.8.8:53
2023/07/12 14:37:43 [Info] [4145505862] app/dispatcher: default route for udp:8.8.8.8:53
2023/07/12 14:37:43 [Debug] [4145505862] app/proxyman/outbound: proxying to proxy_a for dest udp:8.8.8.8:53
2023/07/12 14:37:43 [Info] [4145505862] proxy/freedom: connection opened to udp:8.8.8.8:53, local endpoint 0.0.0.0:0, remote endpoint 0.0.0.0:0

**Reply by qianbinbin** [reply]
你是用 VPS 作为中转连接到住宅 IP 吗？按你的需求，所有流量走住宅 IP，那就把 VPS 上的 outbound 设置到你的 socks 代理即可，不需要在 VPS 上配置分流（要是有网站想直连，可以在本地配置分流），如果你是自己手动搭建的 v2ray，应该很简单吧

你的本地和 VPS 配置、log 什么样，都不清楚，我感觉就是在盲人摸象
